File: androidApp/src/main/kotlin/com/snapp/android/ui/widget/WidgetRegistry.kt
package com.snapp.android.ui.widget

import androidx.compose.runtime.Composable

/** Registry: type String → @Composable renderer. See ARCHITECTURE ui/widget/. */
object WidgetRegistry {
    private val builders = mutableMapOf<String, @Composable (WidgetRenderContext) -> Unit>()

    fun register(type: String, builder: @Composable (WidgetRenderContext) -> Unit) {
        builders[type] = builder
    }

    fun get(type: String): (@Composable (WidgetRenderContext) -> Unit)? = builders[type]
}

File: iosApp/iosApp/Widget/WidgetRegistry.swift
import SwiftUI

/// Registry: type String → (WidgetRenderContext) → AnyView. See ARCHITECTURE Widget/.
final class WidgetRegistry {
    static let shared = WidgetRegistry()
    private var builders: [String: (WidgetRenderContext) -> AnyView] = [:]

    func register(_ type: String, builder: @escaping (WidgetRenderContext) -> AnyView) {
        builders[type] = builder
    }

    func get(_ type: String) -> ((WidgetRenderContext) -> AnyView)? {
        builders[type]
    }
}
